cmd/test_server/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+	"roci.dev/replicache-client/repm"
+)
+
+func main() {
+	impl(os.Args[1:], os.Stdin, os.Stdout, os.Stderr, os.Exit)
+}
+
+func impl(args []string, in io.Reader, out, errs io.Writer, exit func(int)) {
+	app := kingpin.New("test-server", "")
+	app.ErrorWriter(errs)
+	app.UsageWriter(errs)
+	app.Terminate(exit)
+
+	port := app.Flag("port", "The port to run on").Default("7002").Int()
+
+	_, err := app.Parse(args)
+	if err != nil {
+		fmt.Fprintln(errs, err.Error())
+		exit(1)
+	}
+
+	storageDir, err := ioutil.TempDir("", "")
+
+	repm.Init(storageDir, "", nil)
+
+	ps := fmt.Sprintf(":%d", *port)
+	fmt.Printf("Listening on %s...\n", ps)
+	var s http.Handler = testServer{repm.Dispatch}
+	http.ListenAndServe(fmt.Sprintf(":%d", *port), s)
+}
+
+type testServer struct {
+	dispatch func(dbName, rpc string, data []byte) (ret []byte, err error)
+}
+
+func (s testServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	fmt.Println(req.URL.String())
+	if req.URL.Path == "/statusz" {
```

Comment:
[help] Why do we need this?

Reply:
The test pings this before running any test functions to make sure the server is up and running.